hashintel/hash
# Property type form submission fix, autocomplete improvements

## 🌟 What is the purpose of this PR?

Fixes an issue whereby submitting the property type form (e.g. to create a new property type) would also submit the surrounding entity type's form if it had any draft changes.

While not nested in the HTML (as the property type form is in a modal), they are in the same React tree.

This also changes two things about the 'select a type' dropdown:
1. Makes it so that its container isn't taking up more width than it does (the click target previously included the empty space to the right)
2. Makes it close on select (since the majority of properties will be single type) – will need to confirm this doesn't mess with #1513 once that is merged

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203179076056209/1203448542966678) _(internal)_

## ❓ How to test this?



1.  Have an entity type with unsaved changes (E.g. toggle 'required')
2. Open the 'create a property type form'
3. Create a new property type, check that it hasn't automatically saved the entity type (incremented the draft version)
